From 117671edb829f9495cbb725a2f38e62389d3fe64 Mon Sep 17 00:00:00 2001
From: Jonathan Doman <jonathan.doman@intel.com>
Date: Thu, 7 Dec 2023 17:03:15 -0800
Subject: [PATCH 1/1] Catch PECIErrors from supportsControl()

In future backend where supportsControl() requires a PECI call, make
sure any calls are inside a try/catch so potential errors are handled.

Change-Id: Ib35030fa4df764a3cccd922099cd76c77823b7c4
Signed-off-by: Jonathan Doman <jonathan.doman@intel.com>
---
 src/speed_select.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/speed_select.cpp b/src/speed_select.cpp
index fb92dd1..3ae89b8 100644
--- a/src/speed_select.cpp
+++ b/src/speed_select.cpp
@@ -253,9 +253,9 @@ class CPUConfig : public BaseCurrentOperatingConfig
             std::cerr << __func__ << ": Failed to get SST provider instance\n";
             return sdbusplus::message::object_path();
         }
-        setPropertyCheckOrThrow(*sst);
         try
         {
+            setPropertyCheckOrThrow(*sst);
             sst->setCurrentLevel(newConfig->level);
             currentLevel = newConfig->level;
         }
-- 
2.25.1

